### PR TITLE
fix(RadioGroupItem): fix custom indicator

### LINKED ISSUE #14434: RadioGroupItem indicator conflicting with the default
### Environment Information

- **Package version(s)**: @fluentui/react-northstar@0.50.0
- **Browser and OS versions**: Chrome

Please provide a reproduction of the bug in a CodeSandBox: https://codesandbox.io/s/goofy-silence-jk6q9?file=/example.js:83-147

#### Actual behavior:

Passing indicator for `RadioGroupItem` both, the default indicator and the custom one, are being shown.

#### Expected behavior:

Only the indicator passed as prop should be rendered. 

### Priorities and help requested:

Are you willing to submit a PR to fix? Yes

Requested priority: Normal

### PR DESCRIPTION
**Pull request by assuncaocharles**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #14434
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

After discussed we decided to use icons as indicator instead having it as css background, this PR set the default indicator prop and adds the `checkedIndicator` prop and rename the `indicatorBorderColorDefault` variable to `indicatorColorDefault`

#### Focus areas to test

(optional)

### CODE REVIEW

File: packages/fluentui/react-northstar/src/themes/teams/components/RadioGroup/radioGroupItemStyles.ts [hunk 1]
```diff
@@ -63,23 +63,23 @@ export const radioGroupItemStyles: ComponentSlotStylesPrepared<RadioGroupItemSty
...
+    color: v.indicatorBorderColorDefault,
 
     ...(p.checked && {
-      background: v.indicatorBackgroundColorChecked,
+      color: v.indicatorBackgroundColorChecked,
       borderColor: 'transparent',
     }),
 
     ...(p.disabled && {
       borderColor: v.colorDisabled,
```
Comment (mnajdova) [inline packages/fluentui/react-northstar/src/themes/teams/components/RadioGroup/radioGroupItemStyles.ts:81]: I don't think we need the borderColor props anymore

File: packages/fluentui/react-northstar/src/themes/teams/components/RadioGroup/radioGroupItemStyles.ts [hunk 2]
```diff
@@ -63,23 +63,23 @@ export const radioGroupItemStyles: ComponentSlotStylesPrepared<RadioGroupItemSty
...
-    borderStyle: 'solid',
-    borderColor: v.indicatorBorderColorDefault,
+    outline: 0,
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
     width: pxToRem(12),
     height: pxToRem(12),
+    verticalAlign: 'midddle',
+    color: v.indicatorBorderColorDefault,
```
Comment (mnajdova) [inline packages/fluentui/react-northstar/src/themes/teams/components/RadioGroup/radioGroupItemStyles.ts:73]: Can we rename these variables to not contain the `border` in their name?

File: packages/fluentui/react-northstar/src/themes/teams/components/RadioGroup/radioGroupItemStyles.ts (hunk 1 above)
Comment (assuncaocharles) [reply packages/fluentui/react-northstar/src/themes/teams/components/RadioGroup/radioGroupItemStyles.ts:81]: Removed

File: packages/fluentui/react-northstar/src/themes/teams/components/RadioGroup/radioGroupItemStyles.ts (hunk 2 above)
Comment (assuncaocharles) [reply packages/fluentui/react-northstar/src/themes/teams/components/RadioGroup/radioGroupItemStyles.ts]: Done

File: packages/fluentui/react-northstar/src/components/RadioGroup/RadioGroupItem.tsx [hunk 3]
```diff
@@ -54,6 +55,9 @@ export interface RadioGroupItemProps extends UIComponentProps, ChildrenComponent
   /** The radio item indicator can be customized. */
   indicator?: ShorthandValue<BoxProps>;
 
+  /** The radio item indicator can be customized. */
```
Comment (pompomon) [inline packages/fluentui/react-northstar/src/components/RadioGroup/RadioGroupItem.tsx:58]:
```suggestion
  /** The selected radio item indicator can be customized. */
```

**Review by pompomon — APPROVED** [review]

**Pull request merged**